src/haversine.jl

```diff
@@ -26,13 +27,17 @@ function (dist::Haversine)(x, y)
     a = sind(Δφ/2)^2 + cosd(φ₁)*cosd(φ₂)*sind(Δλ/2)^2
 
     # distance on the sphere
-    2 * dist.radius * asin( min(√a, one(a)) ) # take care of floating point errors
+    2 * (dist.radius * asin( min(√a, one(a)) )) # take care of floating point errors
 end
 
-haversine(x, y, radius::Number=Float32(6_371_000)) = Haversine(radius)(x, y)
+haversine(x, y, radius::Number=6_371_000) = Haversine(radius)(x, y)
 
 @noinline haversine_error(dist) = throw(ArgumentError("expected both inputs to have length 2 in $dist distance"))
 
+function result_type(::Haversine{T1}, ::Type{T2}, ::Type{T3}) where {T1<:Number,T2<:Number,T3<:Number}
+    float(promote_type(T1, T2, T3))
+end
```

Comment:
Actually, moving to the `function` syntax made the line wider than it needs to be. :-)
```suggestion
result_type(::Haversine{T1}, ::Type{T2}, ::Type{T3}) where {T1<:Number,T2<:Number,T3<:Number} =
    float(promote_type(T1, T2, T3))
```